Refresh ambient architecture doc (#14994)
istio/istio.io: content/en/docs/ambient/architecture/index.md

Author: SG, cc @craigbox defer this as a follow up!

Author: opened https://github.com/istio/istio.io/issues/15005 to track this

test: n/a
---

## Ambient APIs

### Labels

You can use the following labels to enroll your namespace to ambient, enroll your resource to use a waypoint and specify, the capture scope of your waypoint.

> |  Name  | Feature Status | Scope | Description |

Reviewer: The table format can be improved, suggest to use: http://markdowntable.com/ to format it.

Author: I tried to use it, but i think it is harder to view or make change in the source view.  BTW, the table renders fine in preliminary, which is what users would see.

| --- | --- | --- | --- | --- |
| `istio.io/dataplane-mode` | Beta | `Namespace` |  Specifies the data plane mode, valid value: `ambient`. |
| `istio.io/use-waypoint` | Beta | `Namespace` or `Service` or `Pod` or `WorkloadEntry` or `ServiceEntry` | Enrolls your resource to use a given waypoint, valid value: `#none` or `{namespace}/{waypoint-name}` |

> | `istio.io/waypoint-for` | Alpha | `Gateway` | Specifies the waypoint's capture scope, valid value: `service` or `none` or `workload` or `all`. The default value is `service`. |

Author: I didn't add GatewayClass here, I don't think it is implemented yet?  @ilrudie thoughts?

Reviewer 2: Correct, gw class isn't checked yet :(

Author: Thanks both, marking as resolved

### Policy attachment to waypoints

You can attach Layer 7 policies (such as `AuthorizationPolicy`, `RequestAuthentication`, `Telemetry`, `WasmPlugin`, etc) to your waypoint using `targetRef`.

1. To attach the entire waypoint, set `Gateway` as the `targetRef` value. The example below shows how to attach the `viewer` policy
to the waypoint named `waypoint` for the `default` namespace:

{{< text yaml >}}
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: viewer
  namespace: default
spec:

>   targetRef:

Author: updated

>     kind: Gateway

Author: updated manually

the `productpage-viewer` policy to the `productpage` service in the `default` namespace:

{{< text yaml >}}
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: productpage-viewer
  namespace: default
spec:
  targetRef:
    kind: Service
    group: ""
    name: productpage
{{< /text >}}

## Traffic routing

In {{< gloss "ambient" >}}ambient mode{{< /gloss >}}, workloads can fall into 3 categories:
1. **Uncaptured:** this is a standard pod without any mesh features enabled.
1. **Captured:** this is a pod that has traffic intercepted by {{< gloss >}}ztunnel{{< /gloss >}}. A pod's captured mode can be enabled by setting the `istio.io/dataplane-mode=ambient` label on its namespace, which enables all pods' captured mode for that namespace.
1. **Waypoint enabled:** this is a pod that is "Captured" *and* has a {{< gloss "waypoint" >}}waypoint proxy{{< /gloss >}} deployed.
  If a namespace is labeled with `istio.io/use-waypoint` with its default waypoint for the namespace, the waypoint will apply to all pods in the namespace.
  The `istio.io/use-waypoint` label can optionally be set to apply to only a specific service or pod with its desired waypoint.

>   If the `istio.io/use-waypoint` label exists on the namespace and/or service and/or pod, the pod waypoint takes precedence over the service waypoint, which takes precedence over the namespace waypoint.

Author: Is this correct? @ilrudie or @howardjohn

Reviewer 2: Technically, wl and svc attachment are equal weight because they don't actually compete with each other. there is no need to know which overwrites the other. How the traffic is addressed determines which waypoint would be used. service addressed traffic will not use a wl-attached waypoint and likewise wl addressed traffic will not use a svc-attached waypoint.

Author: The prob is what if my namespace says `use-waypoint: A`, and my service has `use-waypoint: B` and my workload has `use-waypoint: C` or if my service has `use-waypoint: #none` and my workload has `use-waypoint: C`?

Author: Cool, marking as resolved

test: n/a
---

## Ambient APIs

### Labels

> You can use the following labels to enroll your namespace to ambient, enroll your resource to use a waypoint and specify, the capture scope of your waypoint.

Reviewer 3: Suggestion for consistent terms.....

"add your [namespace | service | workload | pod ...] to the mesh"
"enable a waypoint for traffic to your [namespace | service | workload | pod ...] "
"control what traffic is sent to the waypoint"

I'd avoid the use of the term 'enroll' as it's harder to conjugate with 'to' making statements harder to form.

Author: cc @MorrisLaw I had enroll earlier because we have the `istioctl x waypoint --enroll-namespace` cmd. thoughts?  `enable` does sound easier to understand.

Author: updated

Reviewer 4: We can call the `--enroll-namespace` command to enable a waypoint for traffic to your namespace or use the `use-waypoint` annotation to enable a waypoint for traffic to a [service | workload | pod ...]. That kind of wording makes sense to me here for readability and better understanding. I like it, good call!

|  Name  | Feature Status | Scope | Description |
| --- | --- | --- | --- | --- |
| `istio.io/dataplane-mode` | Beta | `Namespace` |  Specifies the data plane mode, valid value: `ambient`. |

> | `istio.io/use-waypoint` | Beta | `Namespace` or `Service` or `Pod` | Enrolls your resource to use a given waypoint, valid values: `{waypoint-name}`, `{namespace}/{waypoint-name}`, or `#none` |

Reviewer 3: replace "Enrolls your resource to use a given waypoint, valid values"

"Use a waypoint for traffic to the labelled resource, valid values"

Author: updated!

> | `istio.io/waypoint-for` | Alpha | `Gateway` | Specifies the waypoint's capture scope, valid value: `service` or `none` or `workload` or `all`. This label is optional and the default value is `service`. |

Reviewer 3: If we want to define a reusable term like 'capture scope' when we need to be a bit more verbose

"Specifies the waypoint's 'capture scope', the class of addressed endpoints it is willing to accept, valid values: `service` or `none` or `workload` or `all`. This label is optional and the default value is `service`. |

or we can avoid trying to define the term and rely on the natural use of 'for' to explain

Specifies what types of endpoints the  waypoint will process traffic for, valid values: `service` or `none` or `workload` or `all`. This label is optional and the default value is `service`. |

Author: updated, took your 2nd suggestion :)

> In order for your `istio.io/use-waypoint` label value to be effective, you have to ensure the used waypoint is configured with the corresponding capture scope which is `service` by default. For example, when you label a pod to use a specific waypoint via the `istio.io/use-waypoint` label, the waypoint's capture scope should be `workload` or `all`.

Reviewer 3: If avoiding the term 'capture scope' ...

In order for your `istio.io/use-waypoint` label value to be effective, you have to ensure  waypoint is configured for the endpoint which is using it. By default waypoints accept traffic for `service` endpoints. For example, when you label a pod to use a specific waypoint via the `istio.io/use-waypoint` label, the waypoint should be labelled  `istio.io./waypoint-for' with the values `workload` or `all`.

Author: agreed, updated!

> ### Policy attachment to waypoints

Author: I think attachment to waypoints is important here, I'll add L7

> ### Policy attachment to waypoints

Author: updated!

> You can attach Layer 7 policies (such as `AuthorizationPolicy`, `RequestAuthentication`, `Telemetry`, `WasmPlugin`, etc) to your waypoint using `targetRef`.

Author: I prefer write it starting with a person, which is the reader :)

> - To attach the entire waypoint, set `Gateway` as the `targetRef` value. The example below shows how to attach the `viewer` policy

Reviewer 5: ```suggestion
- To attach L7 policy to the entire waypoint, set `Gateway` as the `targetRef` value. The example below shows how to attach the `viewer` policy
```

> - To attach the entire waypoint, set `Gateway` as the `targetRef` value. The example below shows how to attach the `viewer` policy

Reviewer 3: s/To attach the entire waypoint/To attach to the entire waypoint

Author: updated

to the waypoint named `waypoint` for the `default` namespace:

    {{< text yaml >}}
    apiVersion: security.istio.io/v1beta1
    kind: AuthorizationPolicy
    metadata:
      name: viewer
      namespace: default
    spec:
      targetRefs:
      - kind: Gateway
        group: gateway.networking.k8s.io
        name: waypoint
    {{< /text >}}

> - To attach a given service within the waypoint, set `Service` as the `targetRef` value. The example below shows how to attach

Reviewer 5: ```suggestion
- To attach L7 policy to a specific service within the waypoint, set `Service` as the `targetRef` value. The example below shows how to attach
```

the `productpage-viewer` policy to the `productpage` service in the `default` namespace:

    {{< text yaml >}}
    apiVersion: security.istio.io/v1beta1
    kind: AuthorizationPolicy
    metadata:
      name: productpage-viewer
      namespace: default
    spec:
      targetRefs:
      - kind: Service
        group: ""
        name: productpage
    {{< /text >}}

## Traffic routing

> In {{< gloss "ambient" >}}ambient mode{{< /gloss >}}, workloads can fall into 3 categories:

Reviewer 5: It feels odd to talk about traffic capture _after_ we talk about L7 policy enforcement via waypoints, which requires traffic capture to work.

Feels like this whole section needs to be at the top.

Author: Maybe :) @craigbox said he would reorganize things once contents are in.  It is possible we may want Ambient APIs to have its own page.  I'll defer this organization as a follow up.

> 1. **Uncaptured:** this is a standard pod without any mesh features enabled.

Reviewer 3: I worry 'capture' is a little too technical here.  We could just as easily say

"Out of Mesh"
"In Mesh"
"Waypoint Enabled"

Author: updated

1. **Captured:** this is a pod that has traffic intercepted by {{< gloss >}}ztunnel{{< /gloss >}}. A pod's captured mode can be enabled by setting the `istio.io/dataplane-mode=ambient` label on its namespace, which enables all pods' captured mode for that namespace.

> 1. **Waypoint enabled:** this is a pod that is "Captured" *and* has a {{< gloss "waypoint" >}}waypoint proxy{{< /gloss >}} deployed.

Reviewer 5: ```suggestion
1. **Waypoint enabled:** this is a pod that is "Captured" *and* has a {{< gloss "waypoint" >}}waypoint proxy{{< /gloss >}} deployed. With a waypoint, L7 policy can be enforced for pod traffic. Waypoints are enabled for a workload by setting the `istio.io/use-waypoint` label.
```

Author: I don't think it is correct to jump in to discuss pod/workload so fast given it is not enabled by default and is not the most common scenario.  Most common scenario is to enable waypoint for service.

Reviewer 5: Yeah the last sentence is not meant to refer to workload-attached waypoints directly, but is just intended to be a generic lead into "here are the ways in which waypoints can be enabled for _your workloads_"

Author: Thanks for the clarification. I reworded it to not focus on pod:

```
To enforce L7 policies, add the `istio.io/use-waypoint` label to your resource to use waypoint for the labeled resource.
```

  If a namespace is labeled with `istio.io/use-waypoint` with its default waypoint for the namespace, the waypoint will apply to all pods in the namespace.

>   The `istio.io/use-waypoint` label can optionally be set to apply to only a specific service or pod with its desired waypoint.

Reviewer 3: The `istio.io/use-waypoint` label can also be set on individual services or pods when using a waypoint for the entire namespace is not desired.

Author: like it!

>   If the `istio.io/use-waypoint` label exists on the namespace and service, the service waypoint takes precedence over the namespace waypoint as long as the service waypoint can capture `service` or `all` traffic.

Reviewer 3: If the `istio.io/use-waypoint` label exists on both a namespace and a service, the service waypoint takes precedence over the namespace waypoint as long as the service waypoint can handle `service` or `all` traffic.

Author: Updated

>   If the `istio.io/use-waypoint` label exists on the namespace and pod, the pod waypoint takes precedence over the namespace waypoint as long as the pod waypoint can capture `workload` or `all` traffic.

Reviewer 3: Can elide if merging above comment

Author: updated!

test: n/a
---

## Ambient APIs

### Labels

You can use the following labels to add your resource to the mesh, use a waypoint for traffic to your resource, and control what traffic is sent to the waypoint.

|  Name  | Feature Status | Resource | Description |
| --- | --- | --- | --- | --- |
| `istio.io/dataplane-mode` | Beta | `Namespace` |  Add your resource to the mesh, valid value: `ambient`. |
| `istio.io/use-waypoint` | Beta | `Namespace` or `Service` or `Pod` | Use a waypoint for traffic to the labeled resource for L7 policy enforcement, valid values: `{waypoint-name}`, `{namespace}/{waypoint-name}`, or `#none` |

> | `istio.io/waypoint-for` | Alpha | `Gateway` | Specifies what types of endpoints the waypoint will process traffic for, valid value: `service` or `none` or `workload` or `all`. This label is optional and the default value is `service`. |

Author: I've considered it but thought it may be harder to edit things because some columns have lots of texts!

In order for your `istio.io/use-waypoint` label value to be effective, you have to ensure waypoint is configured for the endpoint which is using it. By default waypoints accept traffic for service endpoints. For example, when you label a pod to use a specific waypoint via the `istio.io/use-waypoint` label, the waypoint should be labeled `istio.io./waypoint-for` with the value `workload` or `all`.

### Layer 7 policy attachment to waypoints

You can attach Layer 7 policies (such as `AuthorizationPolicy`, `RequestAuthentication`, `Telemetry`, `WasmPlugin`, etc) to your waypoint using `targetRef`.

- To attach L7 policy to the entire waypoint, set `Gateway` as the `targetRefs` value. The example below shows how to attach the `viewer` policy
to the waypoint named `waypoint` for the `default` namespace:

    {{< text yaml >}}
    apiVersion: security.istio.io/v1beta1
    kind: AuthorizationPolicy
    metadata:
      name: viewer
      namespace: default
    spec:
      targetRefs:
      - kind: Gateway
        group: gateway.networking.k8s.io
        name: waypoint
    {{< /text >}}

> - To attach L7 policy to a specific service within the waypoint, set `Service` as the `targetRefs` value. The example below shows how to attach

Reviewer 6: ```suggestion
- To attach a L7 policy to a specific service within the waypoint, set `Service` as the `targetRefs` value. The example below shows how to attach
```

the `productpage-viewer` policy to the `productpage` service in the `default` namespace:

    {{< text yaml >}}
    apiVersion: security.istio.io/v1beta1
    kind: AuthorizationPolicy
    metadata:
      name: productpage-viewer
      namespace: default
    spec:
      targetRefs:
      - kind: Service
        group: ""
        name: productpage
    {{< /text >}}

## Traffic routing

In {{< gloss "ambient" >}}ambient mode{{< /gloss >}}, workloads can fall into 3 categories:
1. **Out of Mesh:** this is a standard pod without any mesh features enabled.

> 1. **In Mesh:** this is a pod that has traffic intercepted at the Layer 4 level by {{< gloss >}}ztunnel{{< /gloss >}}. In this mode, L4 policy can be enforced for pod traffic. This mode can be enabled for a pod by setting the `istio.io/dataplane-mode=ambient` label on the pod's namespace. This will enable *in mesh* mode for all pods in that namespace.

Reviewer 6: ```suggestion
1. **In Mesh:** this is a pod that has traffic intercepted at the Layer 4 level by {{< gloss >}}ztunnel{{< /gloss >}}. In this mode, L4 policies can be enforced for pod traffic. This mode can be enabled for a pod by setting the `istio.io/dataplane-mode=ambient` label on the pod's namespace. This will enable *in mesh* mode for all pods in that namespace.
```

> 1. **Waypoint enabled:** this is a pod that is "In Mesh" *and* has a {{< gloss "waypoint" >}}waypoint proxy{{< /gloss >}} deployed. To enforce L7 policies, add the `istio.io/use-waypoint` label to your resource to use waypoint for the labeled resource.

Reviewer 6: You didn't uppercase in the prior line. Lower case here for consistency:
```suggestion
1. **Waypoint enabled:** this is a pod that is "in mesh" *and* has a {{< gloss "waypoint" >}}waypoint proxy{{< /gloss >}} deployed. To enforce L7 policies, add the `istio.io/use-waypoint` label to your resource to use waypoint for the labeled resource.
```

  - If a namespace is labeled with `istio.io/use-waypoint` with its default waypoint for the namespace, the waypoint will apply to all pods in the namespace.
  - The `istio.io/use-waypoint` label can also be set on individual services or pods when using a waypoint for the entire namespace is not desired.
  - If the `istio.io/use-waypoint` label exists on both a namespace and a service, the service waypoint takes
  precedence over the namespace waypoint as long as the service waypoint can handle service or all traffic.
  Similarly, a label on a pod will take precedence over a namespace label.

Depending on which category a workload is in, the request path will be different.

### Ztunnel routing

#### Outbound

> When a pod in ambient mesh makes an outbound request, it will be transparently redirected to ztunnel which will determine where and how to forward the request.

Reviewer 6: I'd say either a `pod in ambient mode` or `pod in an ambient mesh`. I just picked one
```suggestion
When a pod in an ambient mesh makes an outbound request, it will be transparently redirected to ztunnel which will determine where and how to forward the request.
```

In general, the traffic routing behaves just like Kubernetes default traffic routing;
requests to a `Service` will be sent to an endpoint within the `Service` while requests directly to a `Pod` IP will go directly to that IP.

However, depending on the destination's capabilities, different behavior will occur.
If the destination is also added in the mesh, or otherwise has Istio proxy capabilities (such as a sidecar), the request will be upgraded to an encrypted {{< gloss "HBONE" >}}HBONE tunnel{{< /gloss >}}.
If the destination has a waypoint proxy, in addition to being upgraded to HBONE, the request will also be forwarded to that waypoint for L7 policy enforcement.

Note that in the case of a request to a `Service`, if the service *has* a waypoint, the request will be sent to its waypoint to enforce L7 policies to the traffic.

> Similarly, in the case of a request to a `Pod` IP, if the pod *has* a waypoint, the request will be sent to its waypoint to enforce L7 policies to the traffic.

Author: @bleggett i think you had a comment earlier to clarify if the use-waypoint label for both service and pod, what happens... It is explained here imho as it is for outbound, not inbound.

Reviewer 5: Yep fair enough - we could make it a bit more explicit but this will do.

Since it is possible to vary the labels associated with pods in a `Deployment` it is technically possible for
some pods to use a waypoint while others do not. Users are generally recommended to avoid this advanced use-case.

#### Inbound

> When a pod in ambient mesh receives an inbound request, it will be transparently redirected to ztunnel.

Reviewer 6: ```suggestion
When a pod in an ambient mesh receives an inbound request, it will be transparently redirected to ztunnel.
```

If the destination is also captured, or otherwise has Istio proxy capabilities (such as a sidecar), the request will be upgraded to an encrypted {{< gloss "HBONE" >}}HBONE tunnel{{< /gloss >}}.
If the destination has a waypoint proxy, in addition to being upgraded to HBONE, the request will instead be forwarded to that waypoint.

> Note that in the case of a request to a `Service`, if the service *has* a waypoint, the request will be sent to its waypoint to enforce service-oriented policies to the traffic.

Reviewer 4: @ilrudie is it true that if a service has a waypoint, the traffic will be sent to the waypoint if and only if the waypoint is capable of handling `service` or `all` traffic? If so, should we clarify that here?

Similarly, below, w.r.t. Pod/Workload traffic.

Author: Hi @MorrisLaw I think what you said is true to my understanding. cc @howardjohn @stevenctl to weigh in as @ilrudie is going out of office for the next few days.

Author: My understanding is derived from the [waypoint capture doc](https://docs.google.com/document/d/1DAM_6SajCyC2hD8wBUraaZWZJqUVYPyemzw0-xpHmdY/edit):

```
The system assumes that any Gateway specified by istio.io/use-waypoint is capable of handling redirected traffic subject to this annotation and the binding specification. Since istio.io/use-waypoint plays a role in configuration generation too, it is useful to disable or more finely control redirection while allowing for testing of the generated config. Setting istio.io/waypoint-for: none we allow for this testing scenario.
```

Reviewer 4: > if user tries to attach a waypoint that is svc-only to a workload that will be more or less a noop

good point

Author: Yep that is what i have observed, which makes sense. basically, if I label my pod use-waypoint: C, the pod won't use waypoint C until waypoint C is configured for workload (or all).

Author: Updated to make this clear, PTAL @MorrisLaw or @ilrudie  !!

Reviewer 4: Lgtm

Author: Thanks, marking as resolved

> Similarly, in the case of a request to a `Pod` IP, if the pod *has* a waypoint, the request will be sent to its waypoint to enforce pod-oriented policies to the traffic.

Reviewer 4: Same as above comment. Especially since the waypoint capture scope defaults to service unless specified otherwise. I'm not sure that `use-waypoint` will work as intended if the waypoint only handles service traffic.

Author: Agreed, if the waypoint capture scope isn't workload, i'd think the label of use-waypoint on the pod won't be useful.

Author: Updated to make this clear, PTAL @MorrisLaw !!

Author: Oops, I am so sorry!!! @MorrisLaw

> If the destination is also captured, or otherwise has Istio proxy capabilities (such as a sidecar), the request will be upgraded to an encrypted {{< gloss "HBONE" >}}HBONE tunnel{{< /gloss >}}.

Reviewer 3: I think we should swap HBONE with TLS here, we can have a separate doc on write protocol, TLS & HBONE but I don't think we're trying to sell 'HBONE" here

Author: Defer to this change as we currently have a glossary for hbone but nothing for TLS.

Author: Could update when https://github.com/istio/istio.io/pull/15004/files is merged from @bleggett

> If the destination is also captured, or otherwise has Istio proxy capabilities (such as a sidecar), the request will be upgraded to an encrypted {{< gloss "HBONE" >}}HBONE tunnel{{< /gloss >}}.

Author: updated

> If the destination has a waypoint proxy, in addition to being upgraded to HBONE, the request will instead be forwarded to that waypoint.

Reviewer 3: s/instead/also

Author: updated

> Note that in the case of a request to a `Service`, if the service *has* a waypoint and the waypoint can capture `service` or `all` traffic, the request will be sent to its waypoint to enforce service-oriented policies to the traffic.

Reviewer 3: I don't think we need to keep qualifying that the waypoint is correctly configured to habdle 'service' or 'all' traffic. We've covered it above so we can elide this here.

Also I would s/service-oriented policies/L7 policies  here and elsewhere

Same for 'pod oriented' below, I think these terms are likely to confuse

Author: updated

Similarly, in the case of a request to a `Pod` IP, if the pod *has* a waypoint and the waypoint can capture `workload` or `all` traffic, the request will be sent to its waypoint to enforce pod-oriented policies to the traffic.

> In the rare case that a `Service` has a mix of waypoint enabled and non-enabled endpoints, some requests would be sent to a waypoint while other requests to the same service would not.

Reviewer 3: This is incorrect, a 'Service' either has a waypoint or it does not atomically

I think you may be conflating the definition of K8S Service with the common parlance for a bunch of related worklaods being a 'service'. This case can only happen for 'pods' and to-workload traffic but it's pretty unlikely. If we were going to say anything it would be...

"Since it is possible to vary the labels associated with pods in a Deployment it is technically possible for some pods to *use* a waypoint while others do not. Users are generally recommended to avoid this advanced use-case:"

Author: Yes I was thinking about the service is labelled use-waypoint and some of the workloads belong to the service are labelled as use a different waypoint or use 0 waypoint.  I think this matters in the old design but not in the current design where it relies on traffic to service or workload to decide if waypoint enabled.

accepted your suggestions!

> Because plaintext requests will have no peer identity when Authorization Policies are evaluated,

Reviewer 3: This feels like an abrupt statement and its odd to lead with an edge case rather than the more common stuff which you cover below.

We should contextualize 'plaintext' as 'traffic from sources not yet in the mesh' to be consistent with our terminology earlier.

Author: Updated the `plaintext` wording. Not exactly sure how to fix the abrupt statement seems okay for me....

a user can set a policy requiring an identity (either *any* identity, or a specific one) to block all plaintext traffic.

When the destination is waypoint enabled, if the source is `captured` by its ztunnel, the ztunnel ensures the request **must** go through
the waypoint where policy is enforced. However, a workload outside of the mesh doesn't know anything about waypoint proxies therefore it sends
requests directly to the destination without going through any waypoint proxy even if the destination is waypoint enabled.
Currently, traffic from sidecars and gateways won't go through any waypoint proxy either and they will be made aware of waypoint proxies
in a future release.

### Waypoint routing

> A waypoint exclusively receives HBONE requests.

Reviewer 5: ```suggestion
A waypoint exclusively receives HBONE requests.
```

TODO this should link back to a HBONE doc

Author: noted when that doc is avail :)

Reviewer 5: Yes this is a TODO for me

Reviewer 5: https://github.com/istio/istio.io/pull/15004 - I'll fix that after this merges.

> Upon receiving a request, the waypoint will ensure it is targeting either a `Pod` that it manages or a `Service` that contains a `Pod` it manages.

Reviewer 3: Upon receiving a request, the waypoint will ensure that the traffic is for a `Pod` or `Service`  which *use* it.

Author: updated!

> For either type of request, the waypoint will enforce policies (such as `AuthorizationPolicy`, `RequestAuthentication`, `WasmPlugin`, `Telemetry`, etc) before forwarding.

Reviewer 5: ```suggestion
For either type of request, the waypoint will enforce L7 policies (such as `AuthorizationPolicy`, `RequestAuthentication`, `WasmPlugin`, `Telemetry`, etc) before forwarding.
```

> For either type of request, the waypoint will enforce policies (such as `AuthorizationPolicy`, `RequestAuthentication`, `WasmPlugin`, `Telemetry`, etc) before forwarding.

Reviewer 3: Not sure what is meant by 'type' of request. I think you covered this in the previous sentence so you could just skip. Alternatively can say something like

"Having accepted the traffic the waypoint will then enforce policies (such as `AuthorizationPolicy`, `RequestAuthentication`, `WasmPlugin`, `Telemetry`, etc) before forwarding.

Author: Updated
